clang/runtime/dpct-rt/include/rng_utils.hpp.inc

```diff
@@ -239,6 +241,448 @@ private:
+  /// \param n The number of random numbers.
+  /// \param mean Mean of normal distribution
+  /// \param stddev Standard deviation of normal distribution.
+  inline void generate_gaussian(float *output, std::int64_t n, float mean,
+                                float stddev) {
+    generate<oneapi::mkl::rng::gaussian<float>>(output, n, mean, stddev);
+  }
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|rng_generator_host_generate_gaussian_double|dpct::rng::host
+// DPCT_DEPENDENCY_BEGIN
+// RngUtils|rng_generator_host_generate
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+  /// Generate double random number(s) with the distribution 'gaussian'.
+  /// \param output The pointer of the first random number.
+  /// \param n The number of random numbers.
+  /// \param mean Mean of normal distribution
+  /// \param stddev Standard deviation of normal distribution.
+  inline void generate_gaussian(double *output, std::int64_t n, double mean,
+                                double stddev) {
+    generate<oneapi::mkl::rng::gaussian<double>>(output, n, mean, stddev);
+  }
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|rng_generator_host_generate_poisson|dpct::rng::host
+// DPCT_DEPENDENCY_BEGIN
+// RngUtils|rng_generator_host_generate
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+  /// Generate unsigned int random number(s) with the distribution 'poisson'.
+  /// \param output The pointer of the first random number.
+  /// \param n The number of random numbers.
+  /// \param lambda Lambda for the Poisson distribution.
+  inline void generate_poisson(unsigned int *output, std::int64_t n,
+                               double lambda) {
+    generate<oneapi::mkl::rng::poisson<unsigned int>>(output, n, lambda);
+  }
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|rng_generator_host_generate_uniform_float|dpct::rng::host
+// DPCT_DEPENDENCY_BEGIN
+// RngUtils|rng_generator_host_generate
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+  /// Generate float random number(s) with the distribution 'uniform'.
+  /// \param output The pointer of the first random number.
+  /// \param n The number of random numbers.
+  inline void generate_uniform(float *output, std::int64_t n) {
+    generate<oneapi::mkl::rng::uniform<float>>(output, n);
+  }
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|rng_generator_host_generate_uniform_double|dpct::rng::host
+// DPCT_DEPENDENCY_BEGIN
+// RngUtils|rng_generator_host_generate
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+  /// Generate double random number(s) with the distribution 'uniform'.
+  /// \param output The pointer of the first random number.
+  /// \param n The number of random numbers.
+  inline void generate_uniform(double *output, std::int64_t n) {
+    generate<oneapi::mkl::rng::uniform<double>>(output, n);
+  }
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|rng_generator_host_skip_ahead|dpct::rng::host
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+  /// Skip ahead several random number(s).
+  /// \param num_to_skip The number of random numbers skipped.
```

Comment:
The number of random numbers to be skipped.

Reply:
fixed